datadog/java_agent.go

```diff
@@ -40,9 +40,7 @@ func (j JavaAgent) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
 		if err := sherpa.CopyFile(artifact, file); err != nil {
 			return libcnb.Layer{}, fmt.Errorf("unable to copy artifact to %s\n%w", file, err)
 		}
-
-		layer.LaunchEnvironment.Appendf("JAVA_TOOL_OPTIONS", " ", "-javaagent:%s", file)
-
+		layer.LaunchEnvironment.Default("BPI_DATADOG_AGENT_PATH", file)
```

Comment:
I think this is the culprit to the failing test. You're removing this env variable, but the test is still looking for it.

Reply:
ugh, should have caught that :/ fixing...